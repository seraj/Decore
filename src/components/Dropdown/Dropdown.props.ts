import ThemeProps from "../../theme/Theme.props";

export default interface DropdownProps
  extends ThemeProps,
    React.HTMLAttributes<HTMLDivElement> {
  isOpen: boolean;
}
