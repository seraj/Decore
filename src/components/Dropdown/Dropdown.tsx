import * as React from "react";
import StyledDropdown from "./styled/StyledDropdown";
import DropdownProps from "./Dropdown.props";
import Button from "../Button";

const Dropdown: React.FC<DropdownProps> = ({ children, title, ...props }) => (
  <StyledDropdown {...props} className={props.isOpen ? "open" : ""}>
    <Button>
      {title} <span className="caret" />
    </Button>
    <ul className="dropdown-menu">{children}</ul>
  </StyledDropdown>
);

export default Dropdown;
