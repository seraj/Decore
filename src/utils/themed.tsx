import * as React from "react";
import * as renderer from "react-test-renderer";
import Decore from "../components/Decore";
import theme from "../theme";

const themed = (component: JSX.Element) =>
  renderer.create(<Decore theme={theme}>{component}</Decore>);

export default themed;
