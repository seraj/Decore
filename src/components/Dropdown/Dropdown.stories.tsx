import { text, boolean, withKnobs } from "@storybook/addon-knobs";
import { storiesOf } from "@storybook/react";
import * as React from "react";
import wInfo from "../../utils/wInfo";
import Dropdown from "./Dropdown";

const stories = storiesOf("Components/Dropdown", module);
stories.addDecorator(withKnobs);

stories.add(
  "default",
  wInfo()(() => (
    <Dropdown
      title={text("Label", "Dropdown")}
      isOpen={boolean("isOpen", false)}
    >
      <li>
        <a>Test</a>
      </li>
      <li>
        <a>Test</a>
      </li>
      <li>
        <a>Test</a>
      </li>
      <li>
        <a>Test</a>
      </li>
    </Dropdown>
  ))
);
