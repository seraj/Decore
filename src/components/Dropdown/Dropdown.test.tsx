import * as React from "react";

import themed from "../../utils/themed";
import Dropdown from "./Dropdown";

describe("Dropdown", () => {
  it("should render correctly", () => {
    const component = themed(
      <Dropdown isOpen={false} title="test">
        Test
      </Dropdown>
    ).toJSON();

    expect(component).toMatchSnapshot();
  });
});
