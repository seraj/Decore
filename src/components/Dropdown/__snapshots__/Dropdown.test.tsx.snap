// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dropdown should render correctly 1`] = `
.c0 {
  position: relative;
  cursor: pointer;
}

.c0.open > .dropdown-menu {
  display: block;
}

.c0 .caret {
  display: inline-block;
  margin-left: 8px;
  margin-top: 4px;
  vertical-align: middle;
  border-top: 4px dashed;
  border-top: 4px solid;
  border-right: 4px solid transparent;
  border-left: 4px solid transparent;
}

.c0 .dropdown-menu {
  position: absolute;
  top: 100%;
  left: 0;
  z-index: 1000;
  display: none;
  float: left;
  min-width: 160px;
  padding: 5px 0;
  margin: 2px 0 0;
  font-size: 1rem;
  text-align: left;
  list-style: none;
  background-color: #ffffff;
  border: 1px solid #E5E5E5;
  border-radius: 0;
}

.c0 .dropdown-menu li {
  cursor: pointer;
}

.c0 .dropdown-menu li:hover {
  background: #E5E5E5;
}

.c0 .dropdown-menu li > a {
  display: block;
  padding: 5px 20px;
  color: #333333;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  padding: 0.4rem 1.2rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  font-weight: bold;
  font-size: 14px;
  text-align: center;
  cursor: pointer;
  -webkit-transition: 0.3s;
  transition: 0.3s;
  border-radius: 0px;
  border: 1.5px solid transparent;
  background: #3abf7c;
  color: #ffffff;
}

.c1 .c2 {
  margin-right: 0.5em;
}

.c1 .c2 svg {
  width: 0.8em;
  fill: #ffffff;
}

.c1:disabled {
  background: #E5E5E5;
  color: theme.colors.mono[500];
  cursor: not-allowed;
  pointer-events: none;
}

.c1.light {
  background: #f5f5f5;
  border-radius: 1px;
  color: #333;
  border: none;
}

.c1:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: #ffffff;
  background-color: #5FDD9D;
}

.c1:hover .c2 svg {
  fill: #ffffff;
}

.c1:active {
  background-color: #1B8B5C;
  box-shadow: none !important;
}

.c1:active .c2 svg {
  fill: #ffffff;
}

<div
  className="c0"
>
  <button
    className="c1"
  >
    test
     
    <span
      className="caret"
    />
  </button>
  <ul
    className="dropdown-menu"
  >
    Test
  </ul>
</div>
`;
